src/main/java/physic/PhysicManager.java

```diff
@@ -0,0 +1,87 @@
+package physic;
+
+import geometry.Point;
+import geometry.Rectangle;
+import object.GameObject;
+import object.GameObjectManager;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+public class PhysicManager {
+
+    private Rectangle field;
+
+    private GameObjectManager gameObjectManager;
+
+    public PhysicManager(Rectangle field, GameObjectManager gameObjectManager) {
+        this.field = field;
+        this.gameObjectManager = gameObjectManager;
+    }
+
+    public void move() {
+        for (GameObject gameObject : gameObjectManager.getGameObjects()) {
+            if (canMove(gameObject)) {
+                gameObject.move();
+            }
+        }
+    }
+
+    public boolean canAddObject(GameObject gameObject) {
+        Rectangle gameObjectBody = gameObject.getBody();
+
+        if (isOutOfGameField(gameObjectBody)) {
+            return false;
+        }
+
+        if (doesIntersectsPermeableGameObjects(gameObjectBody, gameObjectManager.getGameObjects())) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean canMove(GameObject gameObject) {
+        if (gameObject.isPermeable()) {
+            return true;
+        }
+
+        Point speed = gameObject.getSpeed();
```

Comment:
Still a little bit confusing that speed object has Point type. May be we can refactor Point class to Vector? What do you think about this?

Reply:
I hesitate doing that, because Vector is also collection, but more than that I don't have objections